supabase/functions/delete-account/index.ts

```diff
@@ -1,70 +1,53 @@
-// Follow this setup guide to integrate the Deno language server with your editor:
-// https://deno.land/manual/getting_started/setup_your_environment
-// This enables autocomplete, go to definition, etc.
+import { createClient } from 'https://esm.sh/@supabase/supabase-js@2.39.0'
+import { corsHeaders } from '../_shared/cors.ts'
 
-// Setup type definitions for built-in Supabase Runtime APIs
-// import "jsr:@supabase/functions-js/edge-runtime.d.ts"
-import { createClient } from '@supabase/supabase-js'
-import { corsHeaders } from './cors.ts'
-
-Deno.serve(async (req: Request) => {
+Deno.serve(async (req) => {
   if (req.method === 'OPTIONS') {
     return new Response('ok', { headers: corsHeaders })
   }
 
   try {
-    // Create a Supabase client with the Auth context of the logged in user.
     const supabaseClient = createClient(
       Deno.env.get('SUPABASE_URL') ?? '',
       Deno.env.get('SUPABASE_ANON_KEY') ?? '',
-      // Create client with Auth context of the user that called the function.
-      // This way your row-level-security (RLS) policies are applied.
       {
         global: {
-          headers: { Authorization: req.headers.get('Authorization')! },
+          headers: { Authorization: req.headers.get('Authorization') ?? '' },
         },
       }
     )
+    const { data: { user }, error: idError } = await supabaseClient.auth.getUser()
+    if (idError) throw new Error(`Could not get user ID after auth: ${idError.message}`)
+    if (!user) throw new Error('Could not get user ID after auth (user null)')
+    const userId = user.id
 
-    // First get the token from the Authorization header
-    const token = req.headers.get('Authorization')!.replace('Bearer ', '')
-
-    // Now we can get the session or user object
-    const {
-      data: { user },
-    } = await supabaseClient.auth.getUser(token)
-    if (!user) throw new Error('User not found')
-
-    // Create admin client for deletion
     const supabaseAdmin = createClient(
       Deno.env.get('SUPABASE_URL') ?? '',
       Deno.env.get('SUPABASE_SERVICE_ROLE_KEY') ?? ''
     )
 
-    // Delete the user using admin API
-    const { error: deleteError } = await supabaseAdmin.auth.admin.deleteUser(user.id)
-    if (deleteError) throw deleteError
+    const { error: deleteError } = await supabaseAdmin.auth.admin.deleteUser(userId)
```

Comment:
So, something for us to think about here:

If there user has a license, and gets deleted, what happens to the license?

Reply:
Nathan had a good solution here: For the delete account problem, I think we just add a warning to that dialog that their license will be deleted as well. And if they ever do it on accident we can always look at the stripe payment records to see that they bought one and reassign it